### PR TITLE
Update deployment things in pom's

### LINKED ISSUE #3: Change groupId
In the pom, the `groupId` should be `io.greymatter`.

### LINKED ISSUE #4: Change artifactId
In the main pom, the `artifactId` should be `data-nifi-sdk`

### LINKED ISSUE #5: Modify distribution
In the main pom, change the distribution to point to maven central:

```xml
    <distributionManagement>
        <snapshotRepository>
            <id>ossrh</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        </snapshotRepository>
        <repository>
            <id>ossrh</id>
            <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
        </repository>
    </distributionManagement>
```

### LINKED ISSUE #6: Modify Deployment
Modify the deployment profile in the main pom to resemble https://github.com/DecipherNow/bouncycastle-maven-plugin/blob/master/pom.xml#L307

### PR DESCRIPTION
**Pull request by dborncamp**
This should take care of all of the things we need to do to get things deployed.

Closes #3 
Closes #4 
Closes #5 
Closes #6

### CODE REVIEW

File: gmd-sdk/nifi-data-processors/pom.xml [hunk 1]
```diff
@@ -77,12 +77,6 @@
           <artifactId>refined-scalaz_${scala.major.version}</artifactId>
           <version>${refined-scalaz.version}</version>
         </dependency>
-        <dependency>
```
Comment (joshua-rutherford) [inline gmd-sdk/nifi-data-processors/pom.xml:80]: Not sure why this is removed as part of this pull request.  Doesn't seem related to the issues being resolved.

Comment (dborncamp) [reply gmd-sdk/nifi-data-processors/pom.xml:80]: It was just unnecessary as it was added to the parent pom long ago and I am back to modifying that file. But you are correct that it doesn't really fit with the issues being fixed.
